libraries/AP_InertialSensor/AP_InertialSensor_SensorHead.cpp

```diff
@@ -0,0 +1,56 @@
+#include <AP_HAL/AP_HAL.h>
+#include <AP_BoardConfig/AP_BoardConfig.h>
+
+#include "AP_InertialSensor_SensorHead.h"
+
+extern const AP_HAL::HAL& hal;
+
+void InertialSensorMessageHandler::handle(InertialSensorMessage::data_t *data)
+{
+    if (_backend->_sem_ins->take(HAL_SEMAPHORE_BLOCK_FOREVER)) {
+        _backend->_gyro.x = data->gyrox;
+        _backend->_gyro.y = data->gyroy;
+        _backend->_gyro.z = data->gyroz;
+        _backend->_accel.x = data->accelx;
+        _backend->_accel.y = data->accely;
+        _backend->_accel.z = data->accelz;
+        _backend->_last_timestamp = AP_HAL::micros64();
+        _backend->_sem_ins->give();
+    }
+}
+
+AP_InertialSensor_SensorHead::AP_InertialSensor_SensorHead(AP_InertialSensor &imu) :
+    AP_InertialSensor_Backend(imu)
+{
+    // TODO: What is the correct id here? Is 0 ok?
```

Comment:
I guess it depends on whether we want the receiving side (aka Slave) to even be aware of the underlying details of each sensor. Since from what I've read, the state estimation & control code deals with sensors abstracted through the libraries, the Slave should be as well. The only case the Slave would care about the underlying details would be once we add support for configuring the sensor backends via the protocol. Does this sound valid?